tools/integration_tests/sparse_file/sparse_file_test.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Integration tests for sparse file functionality
+package sparse_file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/setup"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	testDirName   = "sparse_file_test"
+	largeFileSize = 100 * 1024 * 1024 // 100MB
+	chunkSize     = 20 * 1024 * 1024  // 20MB
+	smallReadSize = 1 * 1024 * 1024   // 1MB
+)
+
+type sparseFileTest struct {
+	flags         []string
+	storageClient *storage.Client
+	ctx           context.Context
+	testDirPath   string
+	cacheDirPath  string
+	suite.Suite
+}
+
+func (s *sparseFileTest) SetupSuite() {
+	s.ctx = context.Background()
+	var err error
+	s.storageClient, err = client.CreateStorageClient(s.ctx)
+	require.NoError(s.T(), err)
+
+	s.cacheDirPath = path.Join(os.Getenv("HOME"), "cache-dir-sparse-file")
+	s.flags = []string{
+		"--implicit-dirs",
+		"--file-cache-max-size-mb=-1",
+		fmt.Sprintf("--cache-dir=%s", s.cacheDirPath),
+		"--file-cache-enable-sparse-file",
+		fmt.Sprintf("--file-cache-download-chunk-size-mb=%d", chunkSize/(1024*1024)),
+	}
+
+	err = static_mounting.MountGcsfuseWithStaticMounting(s.flags)
+	require.NoError(s.T(), err)
+}
+
+func (s *sparseFileTest) SetupTest() {
+	operations.RemoveDir(s.cacheDirPath)
+	s.testDirPath = client.SetupTestDirectory(s.ctx, s.storageClient, testDirName)
+}
+
+func (s *sparseFileTest) TearDownTest() {
+	setup.SaveGCSFuseLogFileInCaseOfFailure(s.T())
+}
+
+func (s *sparseFileTest) TearDownSuite() {
+	setup.UnmountGCSFuseAndDeleteLogFile(setup.TestDir())
+}
+
+func TestSparseFile(t *testing.T) {
+	suite.Run(t, new(sparseFileTest))
```

Comment:
```suggestion
	// TODO: Enable these tests using conditional t.SkipNow() if integrationTest flag is not set.
	t.SkipNow()
	suite.Run(t, new(sparseFileTest))
```

Making this change temporarily to unblock PR linux tests failure. I will raise another PR to fix it properly.